src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -75,5 +83,49 @@ public DebtPositionDTO saveDebtPosition(DebtPositionDTO debtPositionDTO, Organiz
 
     return debtPositionMapper.mapToDto(savedDebtPosition);
   }
+
+  @Override
+  public PaymentOptionDTO saveNewPaymentOption(PaymentOptionDTO paymentOptionDTO) {
+    Pair<PaymentOption, Map<InstallmentNoPII, Installment>> mappedPaymentOption = paymentOptionMapper.mapToModel(paymentOptionDTO);
+    PaymentOption savedPaymentOption = paymentOptionRepository.save(mappedPaymentOption.getFirst());
+
+    savedPaymentOption.getInstallments().forEach(installmentNoPII -> {
+      Installment mappedInstallment = mappedPaymentOption.getSecond().get(installmentNoPII);
+      mappedInstallment.setPaymentOptionId(savedPaymentOption.getPaymentOptionId());
+      if (StringUtils.isBlank(mappedInstallment.getIud())) {
+        String iud = Utilities.getRandomIUD();
+        mappedInstallment.setIud(iud);
+        installmentNoPII.setIud(iud);
+      }
+      Installment savedInstallment = installmentRepository.save(mappedInstallment);
+
+      mappedInstallment.getTransfers().forEach(transfer -> {
+        transfer.setInstallmentId(savedInstallment.getInstallmentId());
+        transferRepository.save(transfer);
+      });
+    });
+
+    return paymentOptionMapper.mapToDto(savedPaymentOption);
+  }
+
+  @Override
+  public InstallmentDTO saveNewInstallment(InstallmentDTO installmentDTO) {
```

Comment:
cannot we reuse createDebtPosition instead?
```suggestion
  @Override
  @Transactional
  public InstallmentDTO saveNewInstallment(InstallmentDTO installmentDTO) {
```